src/artifacts/contract.rs

```diff
@@ -4,42 +4,56 @@ use crate::artifacts::{
     bytecode::{
         Bytecode, BytecodeObject, CompactBytecode, CompactDeployedBytecode, DeployedBytecode,
     },
-    serde_helpers, DevDoc, Evm, Ewasm, LosslessMetadata, Offsets, StorageLayout, UserDoc,
+    DevDoc, Evm, Ewasm, Offsets, StorageLayout, UserDoc,
 };
 use alloy_json_abi::JsonAbi;
 use alloy_primitives::Bytes;
 use serde::{Deserialize, Serialize};
 use std::{borrow::Cow, collections::BTreeMap};
 
+use super::ZkOrSolcMetadata;
+
 /// Represents a compiled solidity contract
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub struct Contract {
     /// The Ethereum Contract Metadata.
     /// See <https://docs.soliditylang.org/en/develop/metadata.html>
     pub abi: Option<JsonAbi>,
-    #[serde(
-        default,
-        skip_serializing_if = "Option::is_none",
-        with = "serde_helpers::json_string_opt"
-    )]
-    pub metadata: Option<LosslessMetadata>,
+
+    pub metadata: Option<ZkOrSolcMetadata>,
+
+    //TODO: remove?
```

Comment:
These you can request. Here is [the list](https://github.com/matter-labs/era-compiler-solidity/blob/main/src/solc/standard_json/input/settings/selection/file/flag.rs) of output fields we support.